cms/djangoapps/contentstore/views/tests/test_course_index.py

```diff
@@ -330,3 +336,129 @@ def _assert_settings_link_present(response):
 
         self.assertEqual(_get_release_date(response), get_default_time_display(self.course.start))
         _assert_settings_link_present(response)
+
+
+class TestCourseReIndex(CourseTestCase):
+    """
+    Unit tests for the course outline.
+    """
+
+    TEST_INDEX_FILENAME = "test_root/index_file.dat"
+
+    def setUp(self):
+        """
+        Set up the for the course outline tests.
+        """
+
+        super(TestCourseReIndex, self).setUp()
+
+        self.course.start = datetime.datetime(2014, 1, 1, tzinfo=pytz.utc)
+        modulestore().update_item(self.course, self.user.id)
+
+        self.chapter = ItemFactory.create(
+            parent_location=self.course.location, category='chapter', display_name="Week 1"
+        )
+        self.sequential = ItemFactory.create(
+            parent_location=self.chapter.location, category='sequential', display_name="Lesson 1"
+        )
+        self.vertical = ItemFactory.create(
+            parent_location=self.sequential.location, category='vertical', display_name='Subsection 1'
+        )
+        self.video = ItemFactory.create(
+            parent_location=self.vertical.location, category="video", display_name="My Video"
+        )
+
+        self.html = ItemFactory.create(
+            parent_location=self.vertical.location, category="html", display_name="My HTML",
+            data="<div>This is my unique HTML content</div>",
+
+        )
+
+        # create test file in which index for this test will live
+        with open(self.TEST_INDEX_FILENAME, "w+") as index_file:
```

Comment:
Yes, it is solely used only for testing when we need a persistence layer for the bok choy tests to share information between Studio (to add items to the index) and LMS (to show results). In aws and devstack, elasticsearch will be the persistence layer, but bok choy does not run in an environment where elasticsearch is present, so we use a shared file for the persistence layer.

The mock index will persist to the file when the file exists, and will not when the file does not exist.

> On Jan 26, 2015, at 10:36 AM, David Baumgold <notifications@github.com> wrote:
> 
> In cms/djangoapps/contentstore/views/tests/test_course_index.py <https://github.com/edx/edx-platform/pull/6506#discussion_r23538176>:
> 
> > +        )
> > +        self.vertical = ItemFactory.create(
> > +            parent_location=self.sequential.location, category='vertical', display_name='Subsection 1'
> > +        )
> > +        self.video = ItemFactory.create(
> > +            parent_location=self.vertical.location, category="video", display_name="My Video"
> > +        )
> > +
> > +        self.html = ItemFactory.create(
> > +            parent_location=self.vertical.location, category="html", display_name="My HTML",
> > +            data="<div>This is my unique HTML content</div>",
> > +
> > +        )
> > +
> > +        # create test file in which index for this test will live
> > +        with open(self.TEST_INDEX_FILENAME, "w+") as index_file:
> Where is this index file actually used? As far as I can tell, it's created in the setUp and removed in the tearDown, but never referenced anywhere else.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-platform/pull/6506/files#r23538176>.
>